checks.d/tokumx.py

```diff
@@ -356,7 +361,15 @@ def _get_replica_metrics(self, conn, tags, server, status):
                     tags.append('role:primary')
                 else:
                     tags.append('role:secondary')
-                    conn.read_preference = ReadPreference.SECONDARY
+                    self.log.debug("Current replSet member is secondary. "
+                                   "Creating new connection to set read_preference to secondary.")
+                    # need a new connection to deal with replica sets
+                    ssl_params = self._get_ssl_params(instance)
+                    conn = MongoClient(server,
+                                       socketTimeoutMS=DEFAULT_TIMEOUT*1000,
+                                       read_preference=ReadPreference.SECONDARY,
+                                       **ssl_params)
+                    db = conn[db.name]
```

Comment:
Looks like we don't handle the authentication on this connection. We should maybe extract the actual connection creation logic from the `_get_connection` method and use it here.